experimental/kernels/ops.cpp

```diff
@@ -22,27 +23,39 @@ void encoder_forward(Context& ctx, float* out,
     uint32_t C;
   };
   setLogLevel(kError);
-  printf("Creating tensors\n");
-  printf("Creating input tensor\%pn", inp);
-  Tensor input = createTensor(ctx, Shape{b * t}, ki32, inp);
-  printf("Created input tensor\n");
-  Tensor wte_t = createTensor(ctx, Shape{v, c}, kf32, wte);
-  printf("Created wte tensor\n");
-  Tensor wpe_t = createTensor(ctx, Shape{t, c}, kf32, wpe);
-  printf("Created wpe tensor\n");
-  Tensor output = createTensor(ctx, Shape{b * t * c}, kf32);
-  printf("Created tensors\n");
+  // Generate the key of the cache by arguments.
+  std::string key = "encoder_forward_" + std::to_string(B) + "_" + std::to_string(T) + "_" + std::to_string(C);
+  Kernel op;
+  if (ctx.kernelPool.data.find(key) == ctx.kernelPool.data.end()) {
+    Tensor input = createTensor(ctx, Shape{b * t}, ki32);
```

Comment:
This might be out of scope for this PR but eventually there shouldn't be any createTensor operations in ops - they should all be passed in.